rpc/src/retry.rs

```diff
@@ -0,0 +1,148 @@
+use std::{
```

Comment:
Worth adding a tracking issue for this as tech debt to be resolved when retries land upstream.

Sorry that you had to write/port all this.

Reply:
No problem, it was a good learning experience. I've created an issue here: https://github.com/0xPolygonZero/zero-bin/issues/102.